bare_metal_billing/tests/test_billing.py

```diff
@@ -0,0 +1,268 @@
+            ["P1", "P2"],
+            [
+                {
+                    "BM FC430": 24 * 1,
+                    "BM FC830": 24 * 2,
+                },
+                {
+                    "BM GPUA100SXM4": 24 * 7,
+                    "BM FC430": 24 * 5,
+                },
+            ],
+        )
+        output_project_invoices = billing.get_project_invoices(
+            test_usage_data, self.TEST_SU_INFO
+        )
+        self.assertEqual(output_project_invoices, answer_project_invoices)
+
+    def test_unknown_su(self):
+        test_usage_data = self._get_bm_usage_data(
+            ["P1", "P1"],
+            resource_classes=["ChickFilA", "Popeyes"],
+            cpus=[1, 1],  # All enough for 1 SU
+            start_times=[datetime(2000, 1, 1, 0, 0, 0)] * 2,
+            expire_times=[
+                datetime(2000, 1, 2, 0, 0, 0),
+                datetime(2000, 1, 4, 0, 0, 0),
+            ],  # 1 and 3 days
+        )
+        answer_project_invoices = self._get_project_invoice_list(
+            ["P1"], [{"BM UNKNOWN": 24 * 4}]
+        )
+        output_project_invoices = billing.get_project_invoices(
+            test_usage_data, self.TEST_SU_INFO
+        )
+        self.assertEqual(output_project_invoices, answer_project_invoices)
+
+    def test_get_su_type(self):
+        # Test known SUs and unknown
+        test_bm_usage_data = self._get_bm_usage_data(
+            ["P1", "P2", "P3", "P4", "P5"],
+            resource_classes=[
+                "lenovo-sd650nv2-a100",
+                "sd650nv2",
+                "fc430",
+                "fc830",
+                "dummy",
+            ],
+        )
+        answer_su_types = (
+            "BM GPUA100SXM4",
+            "BM GPUA100SXM4",
+            "BM FC430",
+            "BM FC830",
+            "BM UNKNOWN",
+        )
+        for i, project_invoice in enumerate(test_bm_usage_data.root):
+            self.assertEqual(billing._get_su_type(project_invoice), answer_su_types[i])
+
+    def test_get_su_quantity(self):
+        """Case of FC430, GPUA100, Unknown"""
+        test_bm_usage_data = self._get_bm_usage_data(
+            ["P1", "P2", "P3"],
+            cpus=[4, 33, 4],  # A100 has 2 SUs of CPU
+            ram=[1024 * 4, 1024 * 9, 1024 * 10],  # UNKNOWN has 3 SUs of RAM
+            gpus=[0, 1, 0],
+        )
+        test_su_types = ("BM FC430", "BM GPUA100SXM4", "BM UNKNOWN")
+        answer_su_quantities = (1, 2, 3)
+        for i in range(len(test_bm_usage_data.root)):
+            self.assertEqual(
+                billing._get_su_quantity(
+                    test_bm_usage_data.root[i], self.TEST_SU_INFO, test_su_types[i]
+                ),
+                answer_su_quantities[i],
+            )
+
+    def test_get_su_quantity_fc430_with_gpu(self):
+        """What happens if a SU without GPU (i.e like FC430) has GPUs?"""
+        # TODO: Is this even possible? Should we have a test case for this? Warnings?
```

Comment:
@knikolla @naved001 [Mainn already confirmed](https://github.com/CCI-MOC/bare-metal-usage-scripts/pull/3#issuecomment-2749079768) that this shouldn't be possible. Should I still keep this test case?

Reply:
Does the code look at the inspection data of the node to see if has GPUs or not?